packages/react-search-ui/src/SearchProvider.js

```diff
@@ -10,29 +10,46 @@ import defaultA11yMessages from "./A11yNotifications";
  * The SearchProvider primarily holds a reference to the SearchDriver and
  * exposes it to the rest of the application in a Context.
  */
-const SearchProvider = ({ children, config }) => {
-  const [driver, setDriver] = useState(null);
+const SearchProvider = ({ children, config = {}, driver }) => {
+  const [driverInstance, setDriverInstance] = useState(null);
 
   useEffect(() => {
     // This initialization is done inside of useEffect, because initializing the SearchDriver server side
     // will error out, since the driver depends on window. Placing the initialization inside of useEffect
     // assures that it won't attempt to initialize server side.
-    const newDriver = new SearchDriver({
-      ...config,
-      a11yNotificationMessages: {
-        ...defaultA11yMessages,
-        ...config.a11yNotificationMessages
-      }
-    });
-    setDriver(newDriver);
+    const currentDriver =
+      driver ||
+      new SearchDriver({
+        ...config,
+        a11yNotificationMessages: {
+          ...defaultA11yMessages,
+          ...config.a11yNotificationMessages
+        }
+      });
+    setDriverInstance(currentDriver);
+
     return () => {
-      newDriver.tearDown();
+      currentDriver.tearDown();
     };
   }, []);
 
+  // This effect allows users to dynamically update their searchQuery without re-mounting a SearchProvider,
+  // which would be destructive. An example of why this is useful is dynamically updating facets.
+  useEffect(() => {
+    if (driverInstance) {
+      driverInstance.setSearchQuery(config.searchQuery);
+    }
+  }, [config.searchQuery]);
```

Comment:
Would you want to pass `driverInstance` into this too? I've read elsewhere that it's good form to include _all_ variables you reference in a hook here

Reply:
Yeah that makes sense. I think for now it will be functionally equivalent but it would protect us from errors in the future.

Reply:
I'm actually going to leave this as is, because there is a functional difference.

I ONLY want this to run when the searchQuery config is changed from the original value.

By adding `driverInstance`, I'm also running it whenever `driverInstance` changes.

When this component mounts, the first effect runs and ends up calling `setDriverInstance`, which calls this method on mount as well.

Luckily, I have a test for this so it broke when I attempted it 😄.

Reply:
Ha! Glad you found the issue so quick.  I'm going to look more into this for my own curiosity, sorry to send you on a goose chase.